### DISCUSSION TITLE
Taxes webhook failure

### DISCUSSION THREAD

**Discussion by steveninety** [swellstores/community#71] [Help] [answered]
Hello all,

I am using a Cloudflare Worker as a taxes webhook url. Testing my code inside the Cloudflare dashboard returns the following:
![image](https://user-images.githubusercontent.com/93653745/210893793-c2cbf6f3-47ff-44ad-ab75-1ed82cb66cd9.png)

The JSON I return matches the format shown in the docs:
![image](https://user-images.githubusercontent.com/93653745/210893883-89dfc703-c0e3-4d23-8df1-e5ef5796c732.png)

Yet, I keep getting errors inside the Swell dashboard:
![image](https://user-images.githubusercontent.com/93653745/210893991-9ec730bf-3fa3-4314-82ac-a8ee9f0c0d3c.png)

What am I possibly doing wrong? The id is correct. 
Furthermore, how can I debug my this beyond the very limited error notification in the dashboard?

Another related question I have, is: 
Can we use the taxes webhook to return taxes on shipping? Instead of only on a per-item basis.

Thanks a lot for any help!

**Comment by logeshswell** [comment, marked as answer]
@steveninety You need to return the json payload present inside the `body` field and not the entire payload inside `$respond`.

**Example:**
Response Code: 200 OK (_It's already correct_)
Response Body: (_Need to fix_)
```node
{
  items: [
    {
      id: itemId,
      taxes: [
        {
          amount: customAmount
        }
      ]
    }
  ]
}
```

**Comment by steveninety** [comment]
Thank you for the answer! - how did I not see that? Late nights haha!

For anyone wondering how to view the request body that's sent to your taxes webhook endpoint:

1. From your webhook endpoint, return the full incoming request body with a distinct name such as: "taxRequestBody".
2. Then, in the admin dashboard's console, make a get /cart request. 
3. Look for "taxRequestBody"
